### PR TITLE
DOC: update qt related prose

### PR DESCRIPTION
**Pull request by tacaswell**
- added qt5 to backend table
 - updated glossary with new qt urls + qt5

### CODE REVIEW

File: doc/glossary/index.rst [hunk 1]
```diff
@@ -91,13 +91,18 @@ Glossary
...
   Qt4
-      `Qt4 <http://trolltech.com/products/qt/>`__ is the most recent
-      version of Qt cross-platform application framework for desktop
-      and embedded development.
+      `Qt4 <http://doc.qt.io/qt-4.8/index.html>`__ is the previous,
+      but most widely used, version of Qt cross-platform application
+      framework for desktop and embedded development.
+
+  Qt5
+      `Qt5 <http://doc.qt.io/qt-5/index.html>`__ is the previous,
```
Comment (jenshnielsen) [inline doc/glossary/index.rst]: Looks like copy paste ;)

Comment (tacaswell) [inline doc/glossary/index.rst]: Gah, good catch.  Fixed and force-pushed.

**Comment by jenshnielsen** [comment]
Looks good. Could you update the pyqt entry too. The qt3 backend is obviously long gone. 
```
      `pyqt <http://wiki.python.org/moin/PyQt>`_ provides python
      wrappers for the :term:`Qt` widgets library and is required by
      the matplotlib QtAgg and Qt4Agg backends.  Widely used on linux
      and windows; many linux distributions package this as
      'python-qt3' or 'python-qt4'.
```

**Comment by tacaswell** [comment]
@jenshnielsen Done.  I wonder how much of our other documentation is that out dated.

**Comment by jenshnielsen** [comment]
Thanks. Will merge once travis is happy. Im sure there is a lot of stuff out of date

**Pull request merged**